### DISCUSSION TITLE
MODNet model without feature_selection()

### DISCUSSION THREAD

**Discussion by mhlee216** [ppdebreuck/modnet#83] [Q&A] [answered]
How can I use MODNet model without `md.feature_selection(n=-1)`?

`model.fit()` doesn't work because of this error.

`AttributeError: 'MODData' object has no attribute 'optimal_features'`

`md.get_optimal_descriptors = df_with_my_features` didn't solve it either.

I want to use my DataFrame(csv file) as MODData.

**Comment by ml-evs** [comment, marked as answer]
Hi @mhlee216, if I am understanding your problem correctly, I think you can simply set `md.optimal_features` (not `get_optimal_descriptors`) with the column names of the descriptors you want to use within your dataset. If you are doing multi-target learning you may also need to set `md.optimal_features_by_target` to a dictionary with structure `{target: [md.optimal_features]} for target in targets}`.

We could certainly provide a nicer interface for this workflow, but the feature selection aspect is a big part of the MODNet approach.

**Reply by mhlee216** [reply]
Thank you!
